tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ManagedExecutorTest.java

```diff
@@ -427,6 +432,435 @@ public void contextControlsForManagedExecutorBuilder() throws InterruptedExcepti
+        ManagedExecutor executor = ManagedExecutor.builder()
+                .propagated(Buffer.CONTEXT_NAME)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+        try {
+            Buffer.set(new StringBuffer("contextualRunnableOverride-buffer-1"));
+            Label.set("contextualRunnableOverride-label-1");
+
+            Runnable precontextualizedTask1 = labelContext.contextualRunnable(() -> {
+                Assert.assertEquals(Label.get(), "contextualRunnableOverride-label-1",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type not cleared from thread.");
+            });
+
+            Buffer.set(new StringBuffer("contextualRunnableOverride-buffer-2"));
+            Label.set("contextualRunnableOverride-label-2");
+
+            Runnable precontextualizedTask2 = labelContext.contextualRunnable(() -> {
+                Assert.assertEquals(Label.get(), "contextualRunnableOverride-label-2",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type not cleared from thread.");
+            });
+
+            Buffer.set(new StringBuffer("contextualRunnableOverride-buffer-3"));
+            Label.set("contextualRunnableOverride-label-3");
+
+            Runnable normalTask = () -> {
+                Assert.assertEquals(Buffer.get().toString(), "contextualRunnableOverride-buffer-3",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Label.get(), "",
+                        "Context type not cleared from thread.");
+            };
+
+            Future<Integer> future = executor.submit(precontextualizedTask1, 1);
+            Assert.assertEquals(future.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(1),
+                    "Unexpected result of task.");
+
+            CompletableFuture<Void> stage0 = executor.runAsync(precontextualizedTask1);
+            CompletableFuture<Void> stage1 = stage0.thenRunAsync(precontextualizedTask1);
+            CompletableFuture<Void> stage2 = stage0.thenRun(precontextualizedTask2);
+            CompletableFuture<Void> stage3 = stage1.runAfterEither(stage2, precontextualizedTask2);
+            CompletableFuture<Void> stage4 = stage1.runAfterBothAsync(stage2, precontextualizedTask1);
+            CompletableFuture<Void> stage5 = stage4.runAfterBoth(stage3, normalTask);
+            stage5.join();
+
+            LinkedBlockingQueue<String> results = new LinkedBlockingQueue<String>();
+            Runnable precontextualizedTask3 = labelContext.contextualRunnable(() -> results.add(Label.get()));
+
+            Buffer.set(new StringBuffer("contextualRunnableOverride-buffer-3"));
+            Label.set("contextualRunnableOverride-label-3");
```

Comment:
Copy/paste error of "contextualRunnableOverride-*-3"?

Reply:
Good catch, this should have switched the values to something different in order to test that the previously captured context is overriding here. I added a commit to fix.